lib/MahoCLI/Commands/SysEncryptionKeyRegenerate.php

```diff
@@ -0,0 +1,244 @@
+                            foreach ($group->fields->children() as $fieldId => $field) {
+                                if ($field->backend_model && (string) $field->backend_model == 'adminhtml/system_config_backend_encrypted') {
+                                    $path = $sectionId . '/' . $groupId . '/' . $fieldId;
+                                    $encryptedPaths[] = $path;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (empty($encryptedPaths)) {
+            $output->writeln('<info>No encrypted configurations to re-encrypt.</info>');
+            return Command::SUCCESS;
+        }
+
+        // Re-encrypting encrypted data on core_config_data
+        $readConnection = Mage::getSingleton('core/resource')->getConnection('core_read');
+        $writeConnection = Mage::getSingleton('core/resource')->getConnection('core_write');
+        $table = Mage::getSingleton('core/resource')->getTableName('core_config_data');
+
+        $select = $readConnection->select()
+            ->from($table)
+            ->where('value IS NOT NULL AND path IN (?)', $encryptedPaths);
+        $encryptedData = $readConnection->fetchAll($select);
+
+        if (empty($encryptedData)) {
+            $output->writeln('<info>No encrypted configurations to re-encrypt.</info>');
+            return Command::SUCCESS;
+        }
+
+        $outputTable = new Table($output);
+        $outputTable->setHeaders(['config_id', 'scope', 'scope_id', 'path']);
+        foreach ($encryptedData as $encryptedDataRow) {
+            $writeConnection->update(
+                $table,
+                ['value' => $this->encrypt($this->decrypt($encryptedDataRow['value']))],
+                ['config_id = ?' => $encryptedDataRow['config_id']],
+            );
+
+            $outputTable->addRow([
+                $encryptedDataRow['config_id'],
+                $encryptedDataRow['scope'],
+                $encryptedDataRow['scope_id'],
+                $encryptedDataRow['path'],
+            ]);
+        }
+
+        $output->writeln('<info>The following configurations were just re-encrypted, make sure to test all of them.</info>');
+        $outputTable->render();
+
+        Mage::dispatchEvent('encryption_key_regenerated', [
+            'output' => $output,
+            'encrypt_callback' => [$this, 'encrypt'],
+            'decrypt_callback' => [$this, 'decrypt'],
+        ]);
+
+        if (\Composer\InstalledVersions::isInstalled('phpseclib/mcrypt_compat')) {
+            $output->writeln('');
+            $output->writeln('<error>Warning: phpseclib/mcrypt_compat is installed. This package can cause encryption issues and should be removed.</error>');
+            $output->writeln('<error>Please remove it using: composer remove phpseclib/mcrypt_compat</error>');
+            $output->writeln('');
+        }
```

Comment:
```suggestion
        if (\Composer\InstalledVersions::isInstalled('mahocommerce/module-mcrypt-compat')) {
            $output->writeln('');
            $output->writeln('<error>You may now remove the compatibility module using: composer remove mahocommerce/module-mcrypt-compat</error>');
            $output->writeln('');
        } elseif (\Composer\InstalledVersions::isInstalled('phpseclib/mcrypt_compat')) {
            $output->writeln('');
            $output->writeln('<error>Warning: phpseclib/mcrypt_compat is installed. This package can cause encryption issues and should be removed.</error>');
            $output->writeln('<error>Please remove it using: composer remove phpseclib/mcrypt_compat</error>');
            $output->writeln('');
        }
```

I had the compatibility module installed, but the message about removing `phpseclib/mcrypt_compat` could be a bit confusing since I didn't directly install that, and thus `composer remove phpseclib/mcrypt_compat` has no effect.

Reply:
ok i'll write a better message there

Reply:
ahhh sorry I read your message but didn't notice the suggestion, what do you think about https://github.com/MahoCommerce/maho/pull/140/commits/c637eba69cee0df4965eedf1810ea52ab0d57a4f?

Reply:
The updated message is good, but I think we should still check for `mahocommerce/module-mcrypt-compat` because once you've generated a new key, the compat module **must** be removed otherwise you'll get this error:

PHP Fatal error:  Uncaught Varien_Exception: Maximum key size must be smaller 56 in /home/www-data/work/maho/git/lib/Varien/Crypt/Mcrypt.php:72

Reply:
totally right, what about this double message? https://github.com/MahoCommerce/maho/pull/140/commits/8791212ae3127514a1fdd6edcfe1efd0d397a8bd

Reply:
Yeah that's perfect. :)